securityscanutils/trivy_scanner_test.go

```diff
@@ -0,0 +1,81 @@
+package securityscanutils_test
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/solo-io/go-utils/osutils/executils"
+	. "github.com/solo-io/go-utils/securityscanutils"
+)
+
+var _ = Describe("Trivy Scanner", func() {
```

Comment:
I like these end to end tests that validate that we can execute trivy directly. I think it could be valuable as well (and the main reason I made the CmdExecutor injectable) to run the trivy scanner with a mock command executor, and confirm that when we return certain status codes and errors, that the trivy scanner only retries a certain number of times. This way we can validate how our TrivyScanner responds to the cmd output.

Reply:
Added!